### ISSUE TITLE
Terraform's latest aws_provider release crashes on localstack

### ISSUE THREAD

**Issue by rayjlinden** [localstack/localstack#3297]
# Type of request: This is a ...

[x] bug report
[ ] feature request

# Detailed description

We use terraform to set up sns/sqs/lambdas.  The latest release 3.19 just broke on localstack.
I filed a bug with terraform - especially since it causes it to crash.  However, I suspect it is now wanting to use a feature you guys may not have implemented yet.  Here is where things go awry when doing the terraform apply:

```
aws_sqs_queue.extraction-fraud-events: Creating...
aws_lambda_function.rebilly_lambda: Creating...
aws_sqs_queue.extraction-payments-events-delay: Creating...
aws_sqs_queue.extraction-payout-events: Creating...
aws_sns_topic.sns-ecom-events: Creating...
aws_lambda_function.receipt_lambda: Creating...
aws_sqs_queue.extraction-payments-events: Creating...
aws_lambda_function.webhook_lambda: Creating...
aws_sns_topic.sns-ecom-events: Creation complete after 1s [id=arn:aws:sns:us-west-2:000000000000:ecom-events-sandbox]
aws_sqs_queue.extraction-fraud-events: Creation complete after 1s [id=http://localhost:4566/000000000000/extraction-fraud-events]
aws_sns_topic_subscription.ecom-events-fraud-target: Creating...
aws_sqs_queue.extraction-payments-events: Creation complete after 1s [id=http://localhost:4566/000000000000/extraction-payments-events]
aws_sns_topic_subscription.ecom-events-payments-target: Creating...
aws_sqs_queue.extraction-payments-events-delay: Creation complete after 1s [id=http://localhost:4566/000000000000/extraction-payments-events-delay]
aws_sns_topic_subscription.ecom-events-payments-target-delay: Creating...
aws_sqs_queue.extraction-payout-events: Creation complete after 1s [id=http://localhost:4566/000000000000/extraction-payout-events]
aws_sns_topic_subscription.ecom-events-payout-target: Creating...
aws_sns_topic_subscription.ecom-events-fraud-target: Creation complete after 0s [id=arn:aws:sns:us-west-2:000000000000:ecom-events-sandbox:76524a11-8f3c-4f14-b3fe-fe0b3d024131]
aws_sns_topic_subscription.ecom-events-payments-target: Creation complete after 0s [id=arn:aws:sns:us-west-2:000000000000:ecom-events-sandbox:74655654-52b8-446a-b320-24960cc97545]
aws_sns_topic_subscription.ecom-events-payments-target-delay: Creation complete after 0s [id=arn:aws:sns:us-west-2:000000000000:ecom-events-sandbox:88405d08-1c97-40f6-8cec-65c18e56beed]
aws_sns_topic_subscription.ecom-events-payout-target: Creation complete after 0s [id=arn:aws:sns:us-west-2:000000000000:ecom-events-sandbox:212f21b0-4e02-4f1d-8115-d8185f1622fc]

Error: rpc error: code = Unavailable desc = transport is closing



Error: rpc error: code = Unavailable desc = transport is closing



Error: rpc error: code = Unavailable desc = transport is closing


panic: runtime error: invalid memory address or nil pointer dereference
2020-12-02T06:46:59.013Z [DEBUG] plugin.terraform-provider-aws_v3.19.0_x5: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4465b2e]
```

## Expected behavior

Things should work.

## Actual behavior

Terraform crashes.

# Steps to reproduce

I could send the terraform script if desired...

## Command used to start LocalStack

Relevant part of a large docker-compose:
```
localstack:
  image: docker.io/localstack/localstack
  environment:
    - SERVICES=sns,sqs,lambda,logs
    - DATA_DIR=/tmp/localstack/data
    - DOCKER_HOST=unix:///var/run/docker.sock
    - DEFAULT_REGION=us-west-2
    - LAMBDA_EXECUTOR=docker
    - AWS_REGION=us-west-2
    - AWS_ACCESS_KEY_ID=localstack
    - AWS_SECRET_ACCESS_KEY=localstack
  ports:
    - "8180:8080" # dashboard
    - "4566:4566" # std port
  volumes:
    - /var/run/docker.sock:/var/run/docker.sock
```

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

N/A

**Comment by rayjlinden** [comment]
I also see a localstack logs a reference to this file:
/tmp/localstack/data/recorded_api_calls.json

Here is the last couple of calls that appear to be made by terraform if that helps:
```
0UXVldWVzUmVzcG9uc2U+"}
{"a": "sqs", "m": "POST", "p": "/", "d": "QWN0aW9uPUxpc3RRdWV1ZXMmVmVyc2lvbj0yMDEyLTExLTA1", "h": {"Remote-Addr": "172.17.0.9", "Host": "76e79af75a41:4566", "Accept-Encoding": "identity", "Content-Type": "application/x-www-form-urlencoded; charset=utf-8", "User-Agent": "Boto3/1.16.26 Python/3.8.2 Linux/4.9.0-8-amd64 Botocore/1.19.26", "X-Amz-Date": "20201202T064914Z", "Authorization": "AWS4-HMAC-SHA256 Credential=__internal_call__/20201202/us-west-2/sqs/aws4_request, SignedHeaders=content-type;host;x-amz-date, Signature=15ddd2a291698433564efe2113ccfeb8610b4ce343c779edb8258c260574ef46", "Content-Length": "36", "X-Forwarded-For": "172.17.0.9, 76e79af75a41:4566, 127.0.0.1, 76e79af75a41:4566", "x-localstack-edge": "https://76e79af75a41:4566", "Connection": "close"}, "rd": "PExpc3RRdWV1ZXNSZXNwb25zZT48TGlzdFF1ZXVlc1Jlc3VsdD48UXVldWVVcmw+aHR0cDovLzc2ZTc5YWY3NWE0MTo0NTY2LzAwMDAwMDAwMDAwMC9leHRyYWN0aW9uLWZyYXVkLWV2ZW50czwvUXVldWVVcmw+PFF1ZXVlVXJsPmh0dHA6Ly83NmU3OWFmNzVhNDE6NDU2Ni8wMDAwMDAwMDAwMDAvZXh0cmFjdGlvbi1wYXltZW50cy1ldmVudHM8L1F1ZXVlVXJsPjxRdWV1ZVVybD5odHRwOi8vNzZlNzlhZjc1YTQxOjQ1NjYvMDAwMDAwMDAwMDAwL2V4dHJhY3Rpb24tcGF5bWVudHMtZXZlbnRzLWRlbGF5PC9RdWV1ZVVybD48UXVldWVVcmw+aHR0cDovLzc2ZTc5YWY3NWE0MTo0NTY2LzAwMDAwMDAwMDAwMC9leHRyYWN0aW9uLXBheW91dC1ldmVudHM8L1F1ZXVlVXJsPjwvTGlzdFF1ZXVlc1Jlc3VsdD48UmVzcG9uc2VNZXRhZGF0YT48UmVxdWVzdElkPkVZRkhDQUNMMldJTVhBTDhQRTRPSlEzTDQzUUNDUkMwQ0NQSFhCOVRVRENKTVc5UzgwV0M8L1JlcXVlc3RJZD48L1Jlc3BvbnNlTWV0YWRhdGE+PC9MaXN0UXVldWVzUmVzcG9uc2U+"}
{"a": "sqs", "m": "POST", "p": "/", "d": "QWN0aW9uPUxpc3RRdWV1ZXMmVmVyc2lvbj0yMDEyLTExLTA1", "h": {"Remote-Addr": "172.17.0.9", "Host": "76e79af75a41:4566", "Accept-Encoding": "identity", "Content-Type": "application/x-www-form-urlencoded; charset=utf-8", "User-Agent": "Boto3/1.16.26 Python/3.8.2 Linux/4.9.0-8-amd64 Botocore/1.19.26", "X-Amz-Date": "20201202T064918Z", "Authorization": "AWS4-HMAC-SHA256 Credential=__internal_call__/20201202/us-west-2/sqs/aws4_request, SignedHeaders=content-type;host;x-amz-date, Signature=678710c284fcb3a2e18cb7dee8bc94a7a58c50dd8e5674b4744f648f4147ecd1", "Content-Length": "36", "X-Forwarded-For": "172.17.0.9, 76e79af75a41:4566, 127.0.0.1, 76e79af75a41:4566", "x-localstack-edge": "https://76e79af75a41:4566", "Connection": "close"}, "rd": "PExpc3RRdWV1ZXNSZXNwb25zZT48TGlzdFF1ZXVlc1Jlc3VsdD48UXVldWVVcmw+aHR0cDovLzc2ZTc5YWY3NWE0MTo0NTY2LzAwMDAwMDAwMDAwMC9leHRyYWN0aW9uLWZyYXVkLWV2ZW50czwvUXVldWVVcmw+PFF1ZXVlVXJsPmh0dHA6Ly83NmU3OWFmNzVhNDE6NDU2Ni8wMDAwMDAwMDAwMDAvZXh0cmFjdGlvbi1wYXltZW50cy1ldmVudHM8L1F1ZXVlVXJsPjxRdWV1ZVVybD5odHRwOi8vNzZlNzlhZjc1YTQxOjQ1NjYvMDAwMDAwMDAwMDAwL2V4dHJhY3Rpb24tcGF5bWVudHMtZXZlbnRzLWRlbGF5PC9RdWV1ZVVybD48UXVldWVVcmw+aHR0cDovLzc2ZTc5YWY3NWE0MTo0NTY2LzAwMDAwMDAwMDAwMC9leHRyYWN0aW9uLXBheW91dC1ldmVudHM8L1F1ZXVlVXJsPjwvTGlzdFF1ZXVlc1Jlc3VsdD48UmVzcG9uc2VNZXRhZGF0YT48UmVxdWVzdElkPlpZVlc3S1VPMFJGM1ZBQlUwV0tIQUVFTVRSTVFBOUlOWFMzR0hJNlZQVk1FRzdBU0lUV0g8L1JlcXVlc3RJZD48L1Jlc3BvbnNlTWV0YWRhdGE+PC9MaXN0UXVldWVzUmVzcG9uc2U+"}
{"a": "sqs", "m": "POST", "p": "/", "d": "QWN0aW9uPUxpc3RRdWV1ZXMmVmVyc2lvbj0yMDEyLTExLTA1", "h": {"Remote-Addr": "172.17.0.9", "Host": "76e79af75a41:4566", "Accept-Encoding": "identity", "Content-Type": "application/x-www-form-urlencoded; charset=utf-8", "User-Agent": "Boto3/1.16.26 Python/3.8.2 Linux/4.9.0-8-amd64 Botocore/1.19.26", "X-Amz-Date": "20201202T064922Z", "Authorization": "AWS4-HMAC-SHA256 Credential=__internal_call__/20201202/us-west-2/sqs/aws4_request, SignedHeaders=content-type;host;x-amz-date, Signature=1133848727858e79f110c249277d5e5bd23bf4db7fc6ae521a3499a3943a385b", "Content-Length": "36", "X-Forwarded-For": "172.17.0.9, 76e79af75a41:4566, 127.0.0.1, 76e79af75a41:4566", "x-localstack-edge": "https://76e79af75a41:4566", "Connection": "close"}, "rd": "PExpc3RRdWV1ZXNSZXNwb25zZT48TGlzdFF1ZXVlc1Jlc3VsdD48UXVldWVVcmw+aHR0cDovLzc2ZTc5YWY3NWE0MTo0NTY2LzAwMDAwMDAwMDAwMC9leHRyYWN0aW9uLWZyYXVkLWV2ZW50czwvUXVldWVVcmw+PFF1ZXVlVXJsPmh0dHA6Ly83NmU3OWFmNzVhNDE6NDU2Ni8wMDAwMDAwMDAwMDAvZXh0cmFjdGlvbi1wYXltZW50cy1ldmVudHM8L1F1ZXVlVXJsPjxRdWV1ZVVybD5odHRwOi8vNzZlNzlhZjc1YTQxOjQ1NjYvMDAwMDAwMDAwMDAwL2V4dHJhY3Rpb24tcGF5bWVudHMtZXZlbnRzLWRlbGF5PC9RdWV1ZVVybD48UXVldWVVcmw+aHR0cDovLzc2ZTc5YWY3NWE0MTo0NTY2LzAwMDAwMDAwMDAwMC9leHRyYWN0aW9uLXBheW91dC1ldmVudHM8L1F1ZXVlVXJsPjwvTGlzdFF1ZXVlc1Jlc3VsdD48UmVzcG9uc2VNZXRhZGF0YT48UmVxdWVzdElkPkxaNFNUU00wVzQ1TlJMMFdPRkZEQ041QUo0WERYR05SN0tTR0wxMVExVDRHODdCRFhFT1Y8L1JlcXVlc3RJZD48L1Jlc3BvbnNlTWV0YWRhdGE+PC9MaXN0UXVldWVzUmVzcG9uc2U+"}
```

I have no idea what that stuff is...

**Comment by usmangani1** [comment]
@rayjlinden Thanks for reporting the issue. Can you please share the terraform script using which we can reproduce this issue ?This will help us identify and resolve the issue sonner.Thanks.

**Comment by rayjlinden** [comment]
Ok - I took the specific parts related to this of our set-up and created this repo to show the issue:
https://github.com/lindenlab/example-terraform-localstack

Just run docker-compose up to see the issue.
If you edit, the file localstack.tf and pin the aws_provider to the previous version - the docker-compose will work.

**Comment by rayjlinden** [comment]
Here is the relevant bug reported on terraforn: https://github.com/hashicorp/terraform-provider-aws/issues/16532
They added a comment that may be of use.

**Comment by usmangani1** [comment]
@rayjlinden Thanks for sharing the resources required to reproduce the issue.
This seems to be due to some changes released in the new version of terraform which are creating a nil pointer error.
Looks like the PR is already raised for the same in the terraform repo.

**Comment by whummer** [comment]
Thanks for the updates @rayjlinden @usmangani1 . This should be fixed in #3302 - can you please pull the `latest` Docker image and give it another try? Please report here if the problem persists. Thanks
